File: src/brs/http/GetAT.java
package brs.http;

import brs.BurstException;
import brs.services.ParameterService;
import com.google.gson.JsonElement;

import javax.servlet.http.HttpServletRequest;

import static brs.http.common.Parameters.AT_PARAMETER;
import static brs.http.common.Parameters.INCLUDE_DETAILS_PARAMETER;

final class GetAT extends APIServlet.JsonRequestHandler {

  private final ParameterService parameterService;

  GetAT(ParameterService parameterService) {
    super(new APITag[]{APITag.AT}, AT_PARAMETER, INCLUDE_DETAILS_PARAMETER);
    this.parameterService = parameterService;
  }

  @Override
  JsonElement processRequest(HttpServletRequest req) throws BurstException {
    boolean includeDetails = !("false".equalsIgnoreCase(req.getParameter(INCLUDE_DETAILS_PARAMETER)));
    return JSONData.at(parameterService.getAT(req), includeDetails);
  }

}

File: src/main/kotlin/brs/http/GetAT.kt
package brs.http

import brs.http.common.Parameters.AT_PARAMETER
import brs.services.AccountService
import brs.services.ParameterService
import com.google.gson.JsonElement
import javax.servlet.http.HttpServletRequest

internal class GetAT(private val parameterService: ParameterService, private val accountService: AccountService) : APIServlet.JsonRequestHandler(arrayOf(APITag.AT), AT_PARAMETER) {
    override fun processRequest(request: HttpServletRequest): JsonElement {
        return JSONData.at(parameterService.getAT(request), accountService)
    }
}
